### PR TITLE
Added SetCommFailure variable support

### PR DESCRIPTION
**Pull request by rafale77**

### CODE REVIEW

File: L_ZWay2.lua [hunk 1]
```diff
@@ -1568,6 +1568,12 @@ local function updateChildren (vDevs)
     local status = state and 2 or -1
     local message = state and state .. " failed" or ''
     dev: status_set (status, message)
+    if state then
+      dev: variable_set (SID.HaDevice, "CommFailure", 1)
+      dev: variable_set (SID.HaDevice, "CommFailureTime", os.time())
+    else
+      dev: variable_set (SID.HaDevice, "CommFailure", 0)
+    end
```
Comment (akbooer) [inline L_ZWay2.lua:1576]: I don't think that this implements what you're intending to do.

Several points to note/check:
* these variables will be set every update – it's deperately inefficient
* are you intending to be able to set a watch on any of these variables?
* are you intending that these updates appear in the log?
* AFAIK, CommFailureTime is cleared when the flag itself is cleared?

**Review by akbooer — CHANGES_REQUESTED** [review]

File: L_ZWay2.lua (hunk 1 above)
Comment (rafale77) [reply L_ZWay2.lua:1576]: Yes I am actually setting a watch on the CommFailure to send me phone notifications. That's why the status is not sufficient. It's a good point that they get set at every turn. Maybe need a flag to set them only if they change? It is the same as the "status" attribute though and the flag should apply the same way?

Comment (akbooer) [reply L_ZWay2.lua:1576]: > Maybe need a flag to set them only if they change?

Yes, that's exactly the point.  Whilst` status_set()` is very fast,` variable_set()` is relatively slow (variables have complex structures and methods to allow logging, watching, archiving, updating UI status...)   The status flag, though, is very much like an attribute, and trivial to set or clear.

**Comment by akbooer** [comment]
Edited code incorporated in next development release

**Pull request closed without merging**